openstf/feature_engineering/general.py

```diff
@@ -67,6 +67,10 @@ def remove_extra_feature_columns(
 
     extra_features = [f for f in list(input_data) if f not in features]
 
+    # Do not see "load" as an extra feature as it is no feature
+    if "load" in extra_features:
```

Comment:
I'm not sure if this should be the responsibility of this function. Did this actually happen by the way? This looks a bit like solving a bug/error somewhere else silently. In my opinion both is not a really good idea. But perhaps I'm missing something?

Reply:
The function basically removes all columns that are not in features, "load" is not a feature but it is essential for what we want to predict. I could slice (.iloc[:,1:]) the input of this function such that load is not passed at all. Is that (combined with the order check earlier) a better solution?

Reply:
I think I did not understand the function properly. I really had to read it a couple times to see what was happening. Perhaps we can improve on this to make it more clear. I'm thinking what makes it confusion? I couple things come to mind:

* I'm not sure "extra" is the right English term in this context. I think "additional" is better perhaps. So we could use `remove_additional_feature_columns` for example? Or maybe any of the other terms you already use in the function: `remove_not_requested_features`, `remove_unrequested_features`
* Perhaps the above combined with a better name of the input arguments could make it more clear:
  * `remove_unrequested_features(input_data, requested_features)` is a lot more clear imo compared to the current situation.

Coming back to your answer. We run into this "issue" on multiple occasion. We have all features our load and horizon in the same datastructure and we don't know if these exist and what their location is. Ideally we would know all these things are these things would be part of a more strictly defined datastructure.

But, until that is the case I don't think the current implementation is a bad idea. You should probably also check on the "Horizon" column besides the "load" column.

Reply:
Good naming sugestions, I implemented them, good to make a separate issue on the datamodels